docgen/src/guides/prepare-for-v3.md

```diff
@@ -45,6 +45,65 @@ const search = instantsearch({
 search.start();
 ```
 
+## transformData vs transformItems
+
+Since InstantSearch.js first public release, we have provided an option to customize the values used in the widgets. This method was letting you map 1-1 the values with other values. With react-instantsearch, we implemented a slightly different API that allowed for mapping but also completly change the content of the list of values. This API is way more powerful and therefore it will be the one that we want to push forward with.
+
+### Current usage
+
+```js
+search.addWidget(
+  instantsearch.widget.refinementList({
+    container: '#someDomNode',
+    attributeName: 'facet',
+    transformData: function(item) {
+      item.count = 0;
+      return item;
+    },
+  })
+);
+```
+
+Will update the counts of every item to 0.
+
+### New usage
+
+```js
+search.addWidget(
+  instantsearch.widget.refinementList({
+    container: '#someDomNode',
+    attributeName: 'facet',
+    transformItems: function(items) {
+      return items.map(function(item) {
+        item.count = 0;
```

Comment:
Maybe in our example we can use a syntax that does not mute the value?

Reply:
Using the spread operator?

Reply:
Yep! I would also spread the array in the other example to avoid mutations.